use-cases.md

```diff
@@ -0,0 +1,184 @@
+
+#### Deserializing Trace Conext from Incoming Request
+
+Let's assume that we have an HTTP server, and the Trace Context is propagated from the client via HTTP headers, accessible via `request.headers`:
+
+```python
+    context = tracer.trace_context_from_text(
+        trace_context_id=request.headers,
+        trace_attributes=request.headers
+    )
+```
+
+Here we set both arguments of the decoding method to the `headers` map. The Tracer object knows which headers it needs to read in order to reconstruct Trace Context, which comprises a trace context ID and trace attributes.
+
+#### Starting a Server-Side Span
+
+The `context` object above can be `None`, if the Tracer did not find relevant headers in the incoming request, either because the client did not send them, or maybe the client is running with a different tracer implementation (e.g. Zipkin vs. AppDash). In this case the server needs to start a brand new trace, otherwise it should join the existing trace:
+
+```python
+    if context is None:
+        span = tracer.start_trace(operation_name=operation)
+    else:
+        span = tracer.join_trace(operation_name=operation,
+                                 parent_trace_context=context)
+    span.set_tag('http.method', request.method)
+    span.set_tag('http.url', request.full_url)
+```
+
+The `operation` above refers to the name the server wants to give to the Span. For example, if the HTTP request was a POST against `/save_user/123`, the operation name can be set to `post:/save_user/`. OpenTracing API does not dictate how application calls the spans.
+
+The `set_tag` calls are examples of recording additional information in the span about the request.
+
+#### In-Process Request Context Propagation
+
+Request context propagation refers to application's ability to associate a certain _context_ with the incoming request such that this context is accessible in all other layers of the application within the same process. It can be used to provide application layers with access to request-specific values such as the identity of the end user, authorization tokens, and the request's deadline. It can also be used for transporting the current tracing Span.
+
+The methods of request context propagation are outside the scope of the OpenTracing API, but it is worth mentioning them here to better understand the following sections. There are two commonly used methods of propagation:
+
+  1. Explicit propagation - the application code is structured to pass around a certain context object. For example, see https://blog.golang.org/context.
+  1. Implicit propagation - the context is stored in platform-specific storage that allows it to be retrieved from any place in the application. Often used by RPC frameworks by utilizing such mechanisms as thread-local or continuation-local storage, or even global variables (in case of single-threaded processes).
```

Comment:
In my mind the main difference is that explicit propagation is done by the end user, and the business application needs to have all its method accept a context object. With implicit propagation the application does not need to change to have the propagation, because it is done by "magic" in a platform-specific way. Implicit propagation always has a performance penalty, while explicit propagation leaks an infrastructure concern into the application code. And in some languages like Go, implicit propagation is simply not possible (horrible hacks of identifying goroutine by the concatenation of its stack trace notwithstanding).

Open to suggestions on how to clarify all of this.